server/controllers/pastSchedule.js

```diff
@@ -0,0 +1,25 @@
+const { errorHandler, notFound } = require("../middleware/error");
+const Event = require("../models/Event");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /event/schedule/past
+// @desc search for entries in the database and returns the past schedules
+// @access Private
+const pastSchedule = asyncHandler(async (req, res) => {
+    const userURL = req.params.userURL;
+    const schedule = await Event.find( {
+        userURL : userURL,
+        eventDate: {
+            $lt: new Date(Date.now())
+        }
+    });
+    // As schedule return [] if there's no event document for user
+    if(Object.entries(schedule).length === 0){
+        throw new Error("not defined");
+    }
```

Comment:
"not defined" isn't really a good error for front-end user. If there are not schedules in the past an empty array is also an indicator of that.

Reply:
I will take the not defined out and just return the empty array